### PR TITLE
fix: sonar cloud Issues

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> Sonar cloud Security Hotspots fixes 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/util/jwtTokenGenerator.js [hunk 1]
```diff
@@ -6,7 +6,6 @@ function base64url(source) {
 
   // Remove padding equal characters
   encodedSource = encodedSource.replace(/=+$/, '');
```
Comment (anantjain45823) [inline src/util/jwtTokenGenerator.js:8]: It is the most optimal regex we can have for scanning the padded '='. Already tried possessive quantifiers and they are not efficient as this one

File: src/v0/destinations/marketo_bulk_upload/fetchJobStatus.js [hunk 2]
```diff
@@ -250,7 +250,7 @@ const responseHandler = async (event, type) => {
   const startTime = Date.now();
   for (const element of responseArr) {
     // split response by comma but ignore commas inside double quotes
-    const elemArr = element.split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);
+    const elemArr = element.split(/,(?=(?:[^"]*"[^"]*")*[^"]*$)/);
```
Comment (anantjain45823) [inline src/v0/destinations/marketo_bulk_upload/fetchJobStatus.js:253]: Not changing this one as it not used much and is going to be deprecated soon

File: src/v0/util/index.js [hunk 3]
```diff
@@ -1419,7 +1419,7 @@ function generateUUID() {
   }
   return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
     // eslint-disable-next-line no-bitwise
-    const r = (d + Math.random() * 16) % 16 | 0;
+    const r = (d + crypto.randomBytes(1)[0]) % 16 | 0;
```
Comment (koladilip) [inline src/v0/util/index.js:1422]: https://www.geeksforgeeks.org/node-js-crypto-randomuuid-function/

already there is direct function available https://nodejs.org/api/crypto.html#cryptorandomuuidoptions

**Review by koladilip — APPROVED** [review]

**Review by sanpj2292 — DISMISSED** [review]

File: src/warehouse/v0/util.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ const toSnakeCase = (str) => {
     return '';
   }
   return String(str)
-    .replace(/^[^A-Za-z0-9]*|[^A-Za-z0-9]*$/g, '')
+    .replace(/^\W+|\W+$/g, '')
```
Comment (achettyiitr) [inline src/warehouse/v0/util.js:9]: W essentially includes `[a-zA-Z0-9_]` right. It does not match with the previous implementation.

**Review by achettyiitr — APPROVED** [review]

**Review by sanpj2292 — DISMISSED** [review]

**Review by koladilip — DISMISSED** [review]

**Review by koladilip — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**